chips/rp2040/src/pwm.rs

```diff
@@ -0,0 +1,1254 @@
+use kernel::utilities::registers::interfaces::{ReadWriteable, Readable, Writeable};
+use kernel::utilities::registers::{register_bitfields, ReadOnly, ReadWrite, WriteOnly};
+use kernel::utilities::StaticRef;
+use kernel::ErrorCode;
+
+use crate::clocks;
+use crate::gpio::RPGpio;
+
+register_bitfields![u32,
+    CSR [
+        // Enable PWM channel
+        EN OFFSET(0) NUMBITS(1) [],
+        // Enable phase-correct modulation
+        PH_CORRECT OFFSET(1) NUMBITS(1) [],
+        // Invert output A
+        A_INV OFFSET(2) NUMBITS(1) [],
+        // Invert output B
+        B_INV OFFSET(3) NUMBITS(1) [],
+        // PWM slice event selection for fractional clock divider
+        // Default value = FREE_RUNNING (always on)
+        // If the event is different from FREE_RUNNING, then pin B becomes
+        // an input pin
+        DIVMOD OFFSET(4) NUMBITS(2) [
+            // Free-running counting at rate dictated by fractional divider
+            FREE_RUNNING = 0,
+            // Fractional divider operation is gated by the PWM B pin
+            B_HIGH = 1,
+            // Counter advances with each rising edge of the PWM B pin
+            B_RISING = 2,
+            // Counter advances with each falling edge of the PWM B pin
+            B_FALLING = 3
+        ],
+        // Retard the phase of the counter by 1 count, while it is running
+        // Self-clearing. Write a 1, and poll until low. Counter must be running.
+        PH_RET OFFSET(6) NUMBITS(1) [],
+        // Advance the phase of the counter by 1 count, while it is running
+        // Self clearing. Write a 1, and poll until low. Counter must be running.
+        PH_ADV OFFSET(7) NUMBITS(1) []
+    ],
+
+    // DIV register
+    // INT and FRAC form a fixed-point fractional number.
+    // Counting rate is system clock frequency divided by this number.
+    // Fractional division uses simple 1st-order sigma-delta.
+    DIV [
+        FRAC OFFSET(0) NUMBITS(4) [],
+        INT OFFSET(4) NUMBITS(8) []
+    ],
+
+    // Direct access to the PWM counter
+    CTR [
+        CTR OFFSET(0) NUMBITS(16) []
+    ],
+
+    // Counter compare values
+    CC [
+        A OFFSET(0) NUMBITS(16) [],
+        B OFFSET(16) NUMBITS(16) []
+    ],
+
+    // Counter top value
+    // When the value of the counter reaches the top value, depending on the
+    // ph_correct value, the counter will either:
+    // + wrap to 0 if ph_correct == 0
+    // + it starts counting downward until it reaches 0 again if ph_correct == 0
+    TOP [
+        TOP OFFSET(0) NUMBITS(16) []
+    ],
+
+    // Control multiple channels at once.
+    // Each bit controls one channel.
+    CH [
+        CH OFFSET(0) NUMBITS(8) [
+            CH0 = 1,
+            CH1 = 2,
+            CH2 = 4,
+            CH3 = 8,
+            CH4 = 16,
+            CH5 = 32,
+            CH6 = 64,
+            CH7 = 128
+        ]
+    ]
+];
+
+const NUMBER_CHANNELS: usize = 8;
+
+#[repr(C)]
+struct Ch {
```

Comment:
Why not `Channel`? (https://github.com/tock/tock/blob/master/doc/Style.md#using-descriptive-names)

Reply:
Changed the name.